K-evolution/auxiliary_library.py

```diff
@@ -807,15 +835,21 @@ def vectorized_recursive_basis(depth_list, seed_ops_list, Hamiltonian, rho0):
     #else:
     #    raise Exception("Incursive depth parameter must be natural") 
     
+    # Use this is more standard, shorter and  clearer:
+    # assert len(depth_list) == len(seed_ops_list), "Insufficient depth parameters"
```

Comment:
A still better way to ensure this would be to use, instead of two lists for depths and ops, as a parameter a list of tuples
[(depth_1, op_1),(depth_2, op_2)...]